impeller/renderer/backend/vulkan/test/mock_vulkan.h

```diff
@@ -5,17 +5,41 @@
 #pragma once
 
 #include <functional>
+#include <memory>
 #include <string>
 #include <vector>
 
 #include "impeller/renderer/backend/vulkan/context_vk.h"
+#include "vulkan/vulkan_enums.hpp"
 
 namespace impeller {
 namespace testing {
 
 std::shared_ptr<std::vector<std::string>> GetMockVulkanFunctions(
     VkDevice device);
 
+// A test-controlled version of |vk::Fence|.
+class MockFence final {
+ public:
+  MockFence() = default;
+
+  // Returns the result that was set in the constructor or |SetStatus|.
+  VkResult GetStatus() { return static_cast<VkResult>(result_); }
+
+  // Sets the result that will be returned by `GetFenceStatus`.
+  static void SetStatus(vk::UniqueFence& fence, vk::Result result) {
+    // Cast the fence to a MockFence and set the result.
+    VkFence raw_fence = fence.get();
+    MockFence* mock_fence = reinterpret_cast<MockFence*>(raw_fence);
+    mock_fence->result_ = result;
+  }
+
+ private:
+  vk::Result result_ = vk::Result::eSuccess;
```

Comment:
You should make this atomic since potentially people are reading and writing from this at the same time.